### PR TITLE
zeromq: add draft/ws/radix_tree options

### PR DESCRIPTION
**Pull request by toge**
Specify library name and version:  **zeromq/all**

close https://github.com/conan-io/conan-center-index/issues/8377.

---

- [x] I've read the [guidelines](https://github.com/conan-io/conan-center-index/blob/master/docs/how_to_add_packages.md) for contributing.
- [x] I've followed the [PEP8](https://www.python.org/dev/peps/pep-0008/) style guides for Python code in the recipes.
- [x] I've used the [latest](https://github.com/conan-io/conan/releases/latest) Conan client version.
- [x] I've tried at least one configuration locally with the
      [conan-center hook](https://github.com/conan-io/hooks.git) activated.

### CODE REVIEW

**Review by ericLemanissier — APPROVED** [review]

**Review by SSE4 — APPROVED** [review]

File: recipes/zeromq/all/conanfile.py [hunk 1]
```diff
@@ -17,14 +18,20 @@ class ZeroMQConan(ConanFile):
         "fPIC": [True, False],
         "encryption": [None, "libsodium", "tweetnacl"],
         "with_norm": [True, False],
-        "poller": [None, "kqueue", "epoll", "devpoll", "pollset", "poll", "select"]
+        "poller": [None, "kqueue", "epoll", "devpoll", "pollset", "poll", "select"],
+        "with_draft_api": [True, False],
+        "with_websocket": [True, False],
```
Comment (uilianries) [inline recipes/zeromq/all/conanfile.py:23]: does it require websocket library?

Comment (toge) [reply recipes/zeromq/all/conanfile.py:23]: I think zeromq has original implementation for websocket. 
https://github.com/zeromq/libzmq/blob/5eb0b00c875b8406f430573bdaf9eab5d4a0c99e/CMakeLists.txt#L177-L216

**Review by madebr — APPROVED** [review]

**Comment by rikercd** [comment]
Forgive me if I am making a silly mistake here, but I've got `with_draft_api` option to my `conanfile.txt` and am still seeing the ZeroMQ cmake output notify that Draft builds are not being made. This is reflected in the CMakeCache.txt file found in the conan packages that get built on my machine: `ENABLE_DRAFTS:BOOL=False`

**Comment by rikercd** [comment]
Digging more, I believe this is due to ZMQ's absurd block of CMake nastiness, which forces ENABLE_DRAFTS to whatever they think is right based on whether or not this is being built from a git repository:

```
if(EXISTS "${CMAKE_SOURCE_DIR}/.git")
  message(STATUS "Build and install draft classes and methods")
  option(ENABLE_DRAFTS "Build and install draft classes and methods" ON)
else()
  message(STATUS "Not building draft classes and methods")
  option(ENABLE_DRAFTS "Build and install draft classes and methods" OFF)
endif()
```

**Comment by uilianries** [comment]
@rikercd Please, open an issue reporting your case, this PR is already closed.

**Comment by rikercd** [comment]
I've been mucking with this the last couple of days and I'm not sure this is something conan-center can really resolve. I am opening an issue in the cppzmq libzmq repositories.

**Pull request merged**